tests/scripts/test_psa_constant_names.py

```diff
@@ -200,26 +229,35 @@ def parse_header(self, filename):
             for line in lines:
                 self.parse_header_line(line)
 
+    _macro_identifier_re = r'[A-Z]\w+'
+    def generate_undeclared_names(self, expr):
+        for name in re.findall(self._macro_identifier_re, expr):
+            if name not in self.all_declared:
+                yield name
+
+    def accept_test_case_line(self, function, argument):
+        #pylint: disable=unused-argument
+        undeclared = list(self.generate_undeclared_names(argument))
+        if undeclared:
+            raise Exception('Undeclared names in test case', undeclared)
```

Comment:
Should we return `False` instead of throwing an exception?

Reply:
The exception here is intended. This only comes up if there's a test case with data that is not declared in a header. If this ever happens, it probably indicates that the test script need to be updated to add a missing header, or that there's a mistake in the test script logic, or that the intent of the test suite has changed in a way that doesn't agree with the test script. Whatever the reason, it's best that we know that something weird is going on, hence the test failure.